python/ray/data/_internal/execution/operators/hash_shuffle.py

```diff
@@ -671,11 +718,18 @@ def _on_aggregation_done(partition_id: int, exc: Optional[Exception]):
             block_gen = aggregator.finalize.options(
                 **finalize_task_resource_bundle
             ).remote(partition_id)
+            self._metrics.on_task_submitted(
+                self._next_shuffle_tasks_idx + partition_id,
+                RefBundle([], owns_blocks=True),
+            )
+            shuffle_reduce_bar.update(i=0, total=self.num_output_rows_total())
```

Comment:
So this will be using single estimate for both shuffle and finalization tasks. Mixing them up doesn't really make a lot of sense.

What i'm thinking is we should explore is using separate metrics for shuffling and finalization stages (which will be kind of split the operator into 2)